webapp/components/admin_console/compliance_settings.jsx

```diff
@@ -40,19 +40,23 @@ export default class ComplianceSettings extends React.Component {
         $('#save-button').button('loading');
 
         const config = this.props.config;
+        const oldEnable = config.ComplianceSettings.Enable;
         config.ComplianceSettings.Enable = this.refs.Enable.checked;
         config.ComplianceSettings.Directory = ReactDOM.findDOMNode(this.refs.Directory).value;
         config.ComplianceSettings.EnableDaily = this.refs.EnableDaily.checked;
 
         Client.saveConfig(
             config,
             () => {
+                $('#save-button').button('reset');
                 AsyncClient.getConfig();
                 this.setState({
                     serverError: null,
                     saveNeeded: false
                 });
-                $('#save-button').button('reset');
+                if (oldEnable !== config.ComplianceSettings.Enable) {
+                    window.location.reload();
```

Comment:
Yes, it's a workaround for now since the compliance page needs to update
On May 10, 2016 20:16, "enahum" <notifications@github.com> wrote:

> In webapp/components/admin_console/compliance_settings.jsx
> <https://github.com/mattermost/platform/pull/2960#discussion_r62775483>:
>
> >                  AsyncClient.getConfig();
> >                  this.setState({
> >                      serverError: null,
> >                      saveNeeded: false
> >                  });
> > -                $('#save-button').button('reset');
> > +                if (oldEnable !== config.ComplianceSettings.Enable) {
> > +                    window.location.reload();
>
> does this need a full page reload?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/mattermost/platform/pull/2960/files/374a5871f92b847dbd434ffbfb27ba9f2d8a0a2c#r62775483>
>